service/src/test/java/bio/terra/profile/pact/provider/BPMProviderTest.java

```diff
@@ -0,0 +1,137 @@
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.data.jdbc.JdbcRepositoriesAutoConfiguration;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.cache.CacheManager;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.FilterType;
+
+@Tag("provider-test")
+@Provider("bpm-provider")
+@PactBroker
+// for local testing, put any test pacts in the service/pacts folder.
+// then comment out the above line, and uncomment the following line
+// @PactFolder("pacts")
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = Main.class,
+    properties = {
+      "profile.profile-database.initialize-on-start=false",
+      "profile.profile-database.upgrade-on-start=false"
+    })
+@EnableAutoConfiguration(
+    exclude = {DataSourceAutoConfiguration.class, JdbcRepositoriesAutoConfiguration.class})
+@ComponentScan(
+    basePackages = {"bio.terra.profile.app.controller", "bio.terra.profile.service"},
+    excludeFilters = {
+      @ComponentScan.Filter(
+          type = FilterType.ASSIGNABLE_TYPE,
+          value = {BeanConfig.class, ProfileDao.class}),
+      @ComponentScan.Filter(
+          type = FilterType.REGEX,
+          pattern = {"jdbcDialect", "jdbcCustomConversions", "jdbcMappingContext"})
+    })
+@ConfigurationPropertiesScan(basePackages = {"bio.terra.profile"})
+public class BPMProviderTest {
+
+  @LocalServerPort int port;
+
+  @MockBean AzureCrlService azureCrlService;
+  @MockBean GcpCrlService gcpCrlService;
+  @MockBean ProfileDao profileDao;
+  @MockBean SamService samService;
+  @MockBean JobService jobService;
+
+  @MockBean ProfileStatusService profileStatusService;
+  @MockBean CacheManager cacheManager;
+  @MockBean UnauthenticatedApiController unauthenticatedApiController;
+
+  @MockBean AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  @Mock AuthenticatedUserRequest userRequest;
+
+  @BeforeEach
+  void setUp(PactVerificationContext context) {
+    when(authenticatedUserRequestFactory.from(any())).thenReturn(userRequest);
+    context.setTarget(new HttpTestTarget("localhost", port, "/"));
+  }
+
+  @TestTemplate
+  @ExtendWith(PactVerificationSpringProvider.class)
+  void verifyPact(PactVerificationContext context) {
+    context.verifyInteraction();
+  }
+
+  @State("a GCP billing profile")
+  Map<String, Object> gcpBillingProfileState() throws InterruptedException {
```

Comment:
Is it possible to have both Azure and GCP billing profiles in BPM's state? From what I can tell, it's either one or the other at the moment

Reply:
At present, that is true.
It _is_ possible to specify multiple states in a single pact (and it's also useful), so that is something I think we need to figure out.

Since the focus on this was the infrastructure integration, and working spring setup, I figured that this was adequate for the current tests to pass, and I didn't want to get bogged down in the details of figuring out mockito setup, and exactly which pact hook is the best place to do that.